plugins/common/kafka/config.go

```diff
@@ -76,14 +78,24 @@ func (k *Config) SetConfig(config *sarama.Config) error {
 
 	config.Producer.Compression = sarama.CompressionCodec(k.CompressionCodec)
 
+	if k.EnableTLS != nil && *k.EnableTLS {
+		config.Net.TLS.Enable = true
+	}
+
 	tlsConfig, err := k.ClientConfig.TLSConfig()
 	if err != nil {
 		return err
 	}
 
 	if tlsConfig != nil {
 		config.Net.TLS.Config = tlsConfig
-		config.Net.TLS.Enable = true
+
+		// To maintain backwards compatibility, if the enable_tls option is not
+		// set TLS is enabled if a non-default TLS config is used.
+		if k.EnableTLS == nil {
+			k.Log.Warnf("Use of deprecated configuration: enable_tls should be set when using TLS")
```

Comment:
```suggestion
```

I do not like calling this a deprecated configuration. Nor do I think this warning should even exist. In any other plugin, if a user sets the TLS config values, then TLS is considered enabled, and I think we can take that same approach here as you have already done.

Reply:
This is fine by me.  It was included only because it existed in previous releases with the prior behavior.

The other plugins I'm looking at seem to only use the TLS config if enable_tls is set to true, but I don't see any reason for a single plugin to have different warning behavior.  I'll also adjust the test that was modified to support the logging.